src/ui/settings/index.js

```diff
@@ -24,20 +24,26 @@ export default function setupSettings(kbnServer, server, config) {
   }
 
   function userSettingsNotFound(kibanaVersion) {
-    if (server.plugins.elasticsearch.status.state === 'green') {
-      server.plugins.kibana.status.red(`Could not find user-provided settings for this version of Kibana (${kibanaVersion})`);
-    } else {
-      server.log(['warning', 'settings'], 'User-provided settings were requested before the Kibana index was ready');
-    }
+    server.plugins.kibana.status.yellow(`Could not find user-provided settings for this version of Kibana (${kibanaVersion})`);
     return {};
   }
 
+  function resetKibanaPluginStatusIfNecessary(user) {
+    const isElasticsearchPluginGreen = server.plugins.elasticsearch.status.state === 'green';
+    const isKibanaPluginCurrentlyYellow = server.plugins.kibana.status.state === 'yellow';
+    if (isElasticsearchPluginGreen && isKibanaPluginCurrentlyYellow) {
```

Comment:
What if `server.plugins.kibana.status.state === 'red'` ?  Does code somewhere else change it from red to yellow, and then this changes it yellow to green?